src/main/java/com/github/saintukrainian/patientdashboard/controller/LaboratoryResultsController.java

```diff
@@ -25,34 +31,59 @@ public String patientLabResults(@RequestParam(required = false) Long patientId,
     if (isNull(patientId)) {
       return "redirect:/";
     }
-    model.addAttribute("labResults",
-        labResultsService.findLabResultsByPatientId(patientId));
-    model.addAttribute("patientId", patientId);
+    model.addAttribute(LAB_RESULTS, labResultsService.findLabResultsByPatientId(patientId));
+    model.addAttribute(PATIENT_ID, patientId);
     return "patient_lab_results.html";
   }
 
+  @GetMapping("/add-labs")
+  public String addLabResults(@RequestParam Long patientId, @ModelAttribute LabResults labResults,
+      Model model) {
+    model.addAttribute(PATIENT_ID, patientId);
+    return "add-labs";
+  }
+
   @GetMapping("/remove-labs")
   public String removeLabResults(@RequestParam Long labsId, @RequestParam Long patientId) {
     labResultsService.removeLabResultsById(labsId);
     return "redirect:/lab-results?patientId=" + patientId;
   }
 
   @GetMapping("/edit-labs")
-  public String editLabResults(@RequestParam Long labsId, @RequestParam Long patientId,
+  public String editLabResults(
+      @RequestParam Long labsId, @RequestParam Long patientId,
       Model model) {
-    model.addAttribute("labResults",
+    model.addAttribute(LAB_RESULTS,
         labResultsService.findLabResultsByLabResultsId(labsId));
-    model.addAttribute("patientId", patientId);
+    model.addAttribute(PATIENT_ID, patientId);
     return "edit-labs";
   }
 
+  @PostMapping("/add-labs")
+  public String addLabResults(
+      @RequestParam(required = false) Long patientId,
+      @Valid @ModelAttribute LabResults labResults, BindingResult bindingResult, Model model) {
+    log.info("Creating new lab results for patient with id = {}", patientId);
```

Comment:
Better put these logs on repository level, because controller methods do not actually do this work, they just delegate it